docs/table_design/Data_distribution.md

```diff
@@ -1,107 +1,209 @@
-    city_code VARCHAR(100),
-    user_name VARCHAR(32) DEFAULT ''
-)
-DUPLICATE KEY(event_day,site_id,pv); -- The data distribution method is not specified.
-```
+Range partitioning is suitable for storing simple, contiguous data, such as time series data (dates or timestamps), or continuous numerical data. And you frequently query and manage data based on continuous date/numerical ranges. Also, it can be applied in some special cases where historical data needs to be partitioned by month, and recent data needs to be partitioned by day.
 
-Specify Hash distribution as the data distribution method at table creation.
+StarRocks stores data in the corresponding partitions based on the explicit mapping of the explictly defined range for each partition.
 
-```SQL
-CREATE TABLE site_access(
-    event_day DATE,
-    site_id INT DEFAULT '10',
-    city_code SMALLINT,
-    user_name VARCHAR(32) DEFAULT '',
-    pv BIGINT SUM DEFAULT '0'
-)
-AGGREGATE KEY(event_day, site_id, city_code, user_name)
-DISTRIBUTED BY HASH(event_day,site_id); -- Set the bucketing method as Hash bucketing and specify the bucketing key.
-```
+**Dynamic partitioning**
 
-Specify Range + Random distribution as the data distribution method at table creation.
+[Dynamic partitioning](./dynamic_partitioning.md) related properties are configured at table creation. StarRocks automatically creates new partitions in advance and removes expired partitions to ensure data freshness, which implements time-to-live (TTL) management for partitions.
 
-```SQL
-CREATE TABLE site_access(
-    event_day DATE,
-    site_id INT DEFAULT '10', 
-    pv BIGINT DEFAULT '0' ,
-    city_code VARCHAR(100),
-    user_name VARCHAR(32) DEFAULT ''
-)
-DUPLICATE KEY(event_day,site_id,pv)
-PARTITION BY RANGE(event_day) (
-    -- Set the partitioning method as Range partitioning.
-    PARTITION p1 VALUES LESS THAN ("2020-01-31"),
-    PARTITION p2 VALUES LESS THAN ("2020-02-29"),
-    PARTITION p3 VALUES LESS THAN ("2020-03-31")
-); -- The bucketing method is not specified, and StarRocks uses Random bucketing by default.
-```
+Different from the automatic partition creation ability provided by the expression partitioning, dynamic partitioning can only periodically creating new partitions based on the properties. If the new data does not belong to these partitions, an error is returned for the loading job. However, the automatic partition creation ability provided by the expression partitioning can always create corresponding new partitions based on the loaded data.
+
+**Manually create partitions**
 
-Specify Range + Hash distribution as the data distribution method at table creation.
+Using a proper partition key can effectively reduce the amount of data scanned during queries. Currently, only date and integer data types are supported as partitioning columns. In business scenarios, partition keys are typically selected from a data management perspective. Common partitioning columns include dates and locations.
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/e8aad55d-1412-42cd-969b-b8cf5197f78c)
-> only columns of date or integer types can be selected to comprise a partition key